### PR TITLE
refactor: [CCM-9242]: Tooltip editor changes for MFE

### PR DESCRIPTION
**Pull request by jenil-harness**
### Summary

 The issue with the tooltips.
 - Tooltips not working in MFEs -  MFE  cannot use Tooltip provider from NG, the reason is that uicore is not shared.
 - Importing tooltip dict from `ng-tooltip` and passing it to TooltipProvider in MFEs show tooltips, but the interactive editor doesn't work. The reason being we mount the editor in NG UI and maintain dict in the component state which we can't pass down to MFE.
 - For making the interactive editor work in MFE, we can mount the editor component again in MFE (which is not a good idea) or we can pass the entire state of tooltip dict using context (this pr).

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 11 Issues 
* Code issue 11 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11477)

File: src/microfrontends/index.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ export interface ParentContext {
   appStoreContext: React.Context<AppStoreContextProps>
   permissionsContext: React.Context<PermissionsContextProps>
   licenseStoreProvider: React.Context<LicenseStoreContextProps>
```
Comment (swarajcheguri-harness) [inline src/microfrontends/index.ts:72]: need to bump package.json in MFE  to auto publish  package

Comment (jenil-harness) [reply src/microfrontends/index.ts:72]: Done

**Comment by jenil-harness** [comment]
retrigger titlecheck

File: jest.config.js [hunk 2]
```diff
@@ -47,7 +47,8 @@ module.exports = {
     '!src/modules/70-pipeline/components/PipelineStudio/PipelineContext/PipelineContext.tsx',
     '!src/modules/70-pipeline/components/PipelineStudio/ExecutionGraph/**',
     '!src/modules/25-governance/**', // 25-governance will be moved to a separate micro-frontend repository shortly
-    '!src/modules/75-cd/factory/**'
+    '!src/modules/75-cd/factory/**',
+    '!src/framework/tooltip/TooltipContext.tsx'
```
Comment (abhinavrastogi-harness) [inline jest.config.js:51]: why no test coverage for this new context?

Comment (jenil-harness) [reply jest.config.js:51]: Not able to see a useful test case for it.

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/framework/tooltip/TooltipContext.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  Harness.openTooltipEditor = () => setShowTooltipEditor(true)
+
+  const onPreviewDatasetFromLocalStorage = React.useCallback(() => {
+    if (showTooltipEditor) {
+      const fromLocalStorage = localStorage.getItem(PREVIEW_TOOLTIP_DATASET_KEY)
+      if (typeof fromLocalStorage === 'string') {
+        try {
+          const parsed = JSON.parse(fromLocalStorage)
+          const isExpired = Date.now() > parsed?.expiry
+          if (!isExpired) {
```
Comment (knagurski) [inline src/framework/tooltip/TooltipContext.tsx:32]: Only used once, might as well inline
```suggestion
          if (Date.now() < parsed?.expiry) {
```

File: src/framework/tooltip/TooltipContext.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { tooltipDictionary } from '@wings-software/ng-tooltip'
+import { TooltipContextProvider } from '@harness/uicore'
+import { NGTooltipEditorPortal } from 'framework/tooltip/TooltipEditor'
+
+export const TooltipContext = React.createContext<Record<string, any>>({})
+const Harness = (window.Harness = window.Harness || {})
+
+const PREVIEW_TOOLTIP_DATASET_KEY = 'previewTooltipDataset'
+
+export function ToolTipProvider(props: React.PropsWithChildren<unknown>): React.ReactElement {
```
Comment (knagurski) [inline src/framework/tooltip/TooltipContext.tsx:18]: Just curious why you went with `React.X` for things like `createContext`, `PropsWithChildren`, `ReactElement` and `useCallback`, but used a named import for `useState`. Not a problem, just curious.

File: src/framework/tooltip/TooltipContext.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState } from 'react'
+import { tooltipDictionary } from '@wings-software/ng-tooltip'
```
Comment (knagurski) [inline src/framework/tooltip/TooltipContext.tsx:9]: Is this not `@harness` now as well?

**Review by knagurski — DISMISSED** [review]
Just one tiny nitpick and a curiosity. Solid work 👍🏻

File: src/framework/tooltip/TooltipContext.tsx (hunk 4 above)
Comment (jenil-harness) [reply src/framework/tooltip/TooltipContext.tsx:18]: Nice catch, I think it is due to Habit 😂

**Review by vkbansal-harness — APPROVED** [review]

File: src/framework/tooltip/TooltipContext.tsx (hunk 4 above)
Comment (knagurski) [reply src/framework/tooltip/TooltipContext.tsx:18]: I usually go the opposite way and name import everything from react. I have a weird aversion to `x.y` in my code 🤷🏼‍♂️

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**